### PR TITLE
Retrieve policy aliases at tenant level

### PR DESCRIPTION
**Pull request by mentat9**
## Checklist

- [X] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [X] The title of the PR is clear and informative
- [X] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [X] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [X] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by mentat9** [comment]
Note: this isn't a breaking change. No inputs, parameters, types or api versions have changed. This is to fix a bug where a few aliases are missing from the subscription-level query. The correct set of aliases to return from this cmdlet is retrieved from the tenant level.

**Comment by msJinLei** [comment]
@mentat9 Please fix the failed test case.

**Comment by mentat9** [comment]
@msJinLei: test is fixed. Also added test coverage for new object type support.

File: src/Resources/ResourceManager/Implementation/Policy/NewAzurePolicyDefinition.cs [hunk 1]
```diff
@@ -141,17 +141,37 @@ private JToken GetResource()
...
-                    PolicyRule = this.GetObjectFromParameter(this.Policy, nameof(this.Policy)),
-                    Metadata = this.Metadata == null ? null : this.GetObjectFromParameter(this.Metadata, nameof(this.Metadata)),
-                    Parameters = this.Parameter == null ? null : this.GetObjectFromParameter(this.Parameter, nameof(this.Parameter)),
-                    Mode = string.IsNullOrEmpty(this.Mode) ? PolicyDefinitionMode.All : this.Mode
-                }
+                Properties = new PolicyDefinitionProperties()
             };
 
+            var policyObject = this.GetObjectFromParameter(this.Policy, nameof(this.Policy));
+            if (policyObject["policyRule"] != null)
```
Comment (pilor) [inline src/Resources/ResourceManager/Implementation/Policy/NewAzurePolicyDefinition.cs]: Have we had requests for supporting this?  If this starts being supported it should be supported in Set-AzPolicyDefinition as well

Comment (pilor) [inline src/Resources/ResourceManager/Implementation/Policy/NewAzurePolicyDefinition.cs]: Should be "properties.policyRule" (and properties.metadata, etc...) right)

File: src/Resources/Resources/ChangeLog.md [hunk 2]
```diff
@@ -21,6 +21,7 @@
 * Update references in .psd1 to use relative path
 * Fix an issue where template deployment fails to read a template parameter if its name conflicts with some built-in parameter name.
 * Updated policy cmdlets to use new api version 2019-09-01 that introduces grouping support within policy set definitions.
+* Update resource client to new version that retrieves providers and aliases at tenant level
```
Comment (pilor) [inline src/Resources/Resources/ChangeLog.md]: The only change was to retrieve policy aliases at tenant level right?  Not providers for Get-AzResourceProvider

File: src/Resources/ResourceManager/Implementation/Policy/GetAzurePolicyAliasCmdlet.cs [hunk 3]
```diff
@@ -116,12 +115,12 @@ private bool IsMatch(string input, string match)
         private IEnumerable<Provider> GetAllProviders()
         {
             var returnList = new List<Provider>();
-            var tempResult = this.ResourceManagerSdkClient.ResourceManagementClient.Providers.List(expand: "resourceTypes/aliases");
+            var tempResult = this.ResourceManagerSdkClient.ResourceManagementClient.Providers.ListAtTenantScope(expand: "resourceTypes/aliases");
```
Comment (pilor) [inline src/Resources/ResourceManager/Implementation/Policy/GetAzurePolicyAliasCmdlet.cs]: No recorded sessionrecords needed to be updated for tests after making this change?

Comment (mentat9) [reply src/Resources/ResourceManager/Implementation/Policy/GetAzurePolicyAliasCmdlet.cs]: I made a recorded session file and added it. Need to investigate why it isn't showing here.

Comment (mentat9) [reply src/Resources/ResourceManager/Implementation/Policy/GetAzurePolicyAliasCmdlet.cs]: I expected it to, but the tests all passed on playback.

File: src/Resources/ResourceManager/Implementation/Policy/NewAzurePolicyDefinition.cs (hunk 1 above)
Comment (mentat9) [reply src/Resources/ResourceManager/Implementation/Policy/NewAzurePolicyDefinition.cs]: Working from a work item you opened a while back. I think there were requests.

File: src/Resources/Resources/ChangeLog.md (hunk 2 above)
Comment (mentat9) [reply src/Resources/Resources/ChangeLog.md]: Correct. The alias cmdlet gets aliases from the providers call results. I'll reword this to avoid confusion on that point.

File: src/Resources/ResourceManager/Implementation/Policy/NewAzurePolicyDefinition.cs (hunk 1 above)
Comment (mentat9) [reply src/Resources/ResourceManager/Implementation/Policy/NewAzurePolicyDefinition.cs]: I chose a policy format without the property bag (you can see the sample file in this PR). This matches what's displayed by the VSCode extension and the portal (when creating a new policy). It also seems more consistent with the format used for the policyRule-flavored input (the policy rule isn't inside ```"policyRule": {}``` or ```"properties": { "policyRule": {} }```). Since this input comes from users, I figured supporting copy-and-paste from VSCode and the portal was a good idea. Let me know if you think I should change it.

File: src/Resources/ResourceManager/Implementation/Policy/GetAzurePolicyAliasCmdlet.cs (hunk 3 above)
Comment (mentat9) [reply src/Resources/ResourceManager/Implementation/Policy/GetAzurePolicyAliasCmdlet.cs]: The tests for this cmdlet aren't live scenario tests so it doesn't need recordings.

**Comment by msJinLei** [comment]
@mentat9 Please resolve the conflicts

**Comment by mentat9** [comment]
@msJinLei conflicts and checks are fixed.

**Comment by msJinLei** [comment]
@mentat9 
- You need to rebase your commits to the latest master manually to prevent some merge errors.
- Why you have to change / add resource manager dependent to other modules? Most of them don't reply on resource manager SDK directly

**Comment by mentat9** [comment]
Believe me, I didn't want to. Unfortunately, the test projects in several modules are using the resource SDK indirectly via two common test projects (below tools folder). I had to make an explicit reference to 3.0.0 version of the resource SDK to get a clean build (version downgrade caused build breaks).

**Comment by msJinLei** [comment]
> Believe me, I didn't want to. Unfortunately, the test projects in several modules are using the resource SDK indirectly via two common test projects (below tools folder). I had to make an explicit reference to 3.0.0 version of the resource SDK to get a clean build (version downgrade caused build breaks).

Could you try this way
- revert all the changes to csproj other than "src/Resources/ResourceManager/ResourceManager.csproj"
- update the reference to "Microsoft.Azure.Management.ResourceManager" in tools/Common.Netcore.Dependencies.Test.targets from 2.2.0 to 3.0 and see whether the tests can pass.

**Comment by mentat9** [comment]
OK, I rebased on official master, updated the .targets file and removed subsequently unneeded changes to .csproj files. All checks have passed.
